styx-api-service/src/main/java/com/spotify/styx/api/WorkflowResource.java

```diff
@@ -86,11 +94,61 @@ public WorkflowResource(Storage storage, DockerImageValidator dockerImageValidat
             rc -> patchState(arg("cid", rc), arg("wfid", rc), rc.request()))
     );
 
+    final List<Route<AsyncHandler<Response<ByteString>>>> forwardedRoutes = Arrays.asList(
+        Route.async(
+            "POST", BASE + "/<cid>",
+            rc -> createOrUpdateWorkflow(arg("cid", rc), rc)
+        ),
+        Route.async(
+            "DELETE", BASE + "/<cid>/<wfid>",
+            rc -> deleteWorkflow(arg("cid", rc), arg("wfid", rc), rc)
+        )
+    );
+
     return cat(
-        Api.prefixRoutes(routes, V3)
+        Api.prefixRoutes(routes, V3),
+        Api.prefixRoutes(forwardedRoutes, V3)
     );
   }
 
+  private CompletionStage<Response<ByteString>> deleteWorkflow(String cid, String wfid,
+                                                               RequestContext rc) {
+    // TODO: handle workflow crud directly in api service instead of proxying to scheduler
+    return rc.requestScopedClient()
+        .send(rc.request().withUri(schedulerApiUrl("workflows", cid, wfid)));
+  }
+
+  private CompletionStage<Response<ByteString>> createOrUpdateWorkflow(String componentId,
+                                                                       RequestContext rc) {
+    final Optional<ByteString> payload = rc.request().payload();
+    if (!payload.isPresent()) {
+      return CompletableFuture.completedFuture(
+          Response.forStatus(Status.BAD_REQUEST.withReasonPhrase("Missing payload.")));
+    }
+    final WorkflowConfiguration workflowConfig;
+    try {
+      workflowConfig = OBJECT_MAPPER
+          .readValue(payload.get().toByteArray(), WorkflowConfiguration.class);
+    } catch (IOException e) {
+      return CompletableFuture.completedFuture(
+          Response.forStatus(Status.BAD_REQUEST.withReasonPhrase("Invalid payload.")));
```

Comment:
We can improve the error message, I think. Instead of `Invalid payload` we can be more specific, like `Invalid workflow configuration` for example.

Reply:
Might be completely unparsable payload though... 🤔

Reply:
We need a flaky network for this to happen...